### DISCUSSION TITLE
Z8102AX firmwre issue

### DISCUSSION THREAD

**Discussion by ajithkumarhg** [4IceG/ZBT-Z8102AX-M2#4] [General]
I have used this OPEN wrt firmware but after reboot it shutck in boot process, showing green and abmer light but not connecting over lan or not showign any wifi configuration, how i can reset it completly ?

**Comment by 4IceG** [comment]
Hi @ajithkumarhg , 
Did you install my build or version from another user (from the OpenWrt forum)?

If router is not responding, you need to restore it/recover using U-Boot and TFTP ([SerialRecovery.zip](https://github.com/ofmodemsandmen/ROOterSource2305/blob/main/configfiles/addons/SerialRecovery.zip)).

I don't know if there is a recovery manual for this router, I hope the instructions from the link above will be helpful.

**Comment by carp4** [comment]
There are also tools and instructions for serial recovery here (instructional pdf inside the zip): 
http://www.aturnofthenut.com/autobuilds/

The current zip for this router is ZBT-Z8102AX-GO2024-01-03-ABSM.zip but it updates weekly.

**Comment by ajithkumarhg** [comment]
Almost yes and I had issues with 5G network for some provider and I was
unable to run at commands.

Other than that firmware was acting weird.

So now I have used uboot and restored it , but If I want to use open wrt
firmware whats the process?

**Comment by ajithkumarhg** [comment]
do you know how to make work of RM520N-GL AP module which only supported PCIe mode,

**Reply by 4IceG** [reply]
If router does not have a full PCIe port, this may not be possible. I didn't check whether the socket has all the signal lines or whether they were limited to the USB version only.

**Comment by ajithkumarhg** [comment]
I am unable to discover the module using USB or on open wrt router.

It's confusing

**Reply by 4IceG** [reply]
You cannot change the RM520N-GL AP modem to work in USB mode, this modem does not support this mode (trying to switch may brick modem).
I'm afraid that router only supports modems in USB mode, I haven't heard of anyone running modem in PCIe mode.

**Comment by ajithkumarhg** [comment]
It's strange the I received rm520n-gl AP module instead of AA regular one.

**Comment by ajithkumarhg** [comment]
Now I need to fight with seller to replace or refund back

**Comment by ajithkumarhg** [comment]
I'll check with ZBT before I take any decision

**Comment by 4IceG** [comment]
If you received the AP version, let the seller show/prove that this version works with this router.

**Comment by ajithkumarhg** [comment]
Yes I am on the same page
